samples/matter/basic/src/app_task.cpp

```diff
@@ -0,0 +1,289 @@
+		 * connectivity to the service OR subscriptions are not fully established
+		 * THEN blink the LED Off for a short period of time.
+		 *
+		 * If the system has ble connection(s) uptill the stage above, THEN blink the LED at an even
+		 * rate of 100ms.
+		 *
+		 * Otherwise, blink the LED On for a very short time. */
+		if (sHaveServiceConnectivity) {
+			sStatusLED.Set(true);
+		} else if (sIsThreadProvisioned && sIsThreadEnabled) {
+			sStatusLED.Blink(950, 50);
+		} else if (sHaveBLEConnections) {
+			sStatusLED.Blink(100, 100);
+		} else {
+			sStatusLED.Blink(50, 950);
+		}
+
+		sStatusLED.Animate();
+		sUnusedLED.Animate();
+		sUnusedLED_1.Animate();
+		sUnusedLED_2.Animate();
+	}
+}
+
+void AppTask::PostEvent(const AppEvent &event)
+{
+	if (k_msgq_put(&sAppEventQueue, &event, K_NO_WAIT)) {
+		LOG_INF("Failed to post event to app task event queue");
+	}
+}
+
+void AppTask::DispatchEvent(const AppEvent &event)
+{
+	switch (event.Type) {
+	case AppEvent::FunctionPress:
+		FunctionPressHandler();
+		break;
+	case AppEvent::FunctionRelease:
+		FunctionReleaseHandler();
+		break;
+	case AppEvent::FunctionTimer:
+		FunctionTimerEventHandler();
+		break;
+	case AppEvent::StartThread:
+		StartThreadHandler();
+		break;
+	case AppEvent::StartBleAdvertising:
+		StartBLEAdvertisingHandler();
+		break;
+	default:
+		LOG_INF("Unknown event received");
+		break;
+	}
+}
+
+void AppTask::FunctionPressHandler()
+{
+	sAppTask.StartFunctionTimer(kFactoryResetTriggerTimeout);
+	sAppTask.mFunction = TimerFunction::FactoryReset;
+}
+
+void AppTask::FunctionReleaseHandler()
+{
+	if (sAppTask.mFunction == TimerFunction::FactoryReset) {
+		sUnusedLED_2.Set(false);
+		sUnusedLED_1.Set(false);
+		sUnusedLED.Set(false);
+
+		sAppTask.CancelFunctionTimer();
+		sAppTask.mFunction = TimerFunction::NoneSelected;
+		LOG_INF("Factory Reset has been Canceled");
+	}
+}
+
+void AppTask::FunctionTimerEventHandler()
+{
+	if (sAppTask.mFunction == TimerFunction::FactoryReset) {
+		sAppTask.mFunction = TimerFunction::NoneSelected;
+		LOG_INF("Factory Reset triggered");
+		
+		sStatusLED.Set(true);
+		sUnusedLED.Set(true);
+		sUnusedLED_1.Set(true);
+		sUnusedLED_2.Set(true);
+
+		ConfigurationMgr().InitiateFactoryReset();
+	}
+}
+
+void AppTask::StartThreadHandler()
+{
+	if (AddTestPairing() != CHIP_NO_ERROR) {
+		LOG_ERR("Failed to add test pairing");
+	}
+
+	if (!ConnectivityMgr().IsThreadProvisioned()) {
+		StartDefaultThreadNetwork();
+		LOG_INF("Device is not commissioned to a Thread network. Starting with the default configuration");
+	} else {
+		LOG_INF("Device is commissioned to a Thread network");
+	}
+}
+
+void AppTask::StartBLEAdvertisingHandler()
+{
+	if (chip::DeviceLayer::ConnectivityMgr().IsThreadProvisioned()) {
+		LOG_INF("NFC Tag emulation and BLE advertisement not started - device is commissioned to a Thread network.");
+		return;
+	}
+
+#ifdef CONFIG_CHIP_NFC_COMMISSIONING
```

Comment:
As far as I remember, we assumed to not support NFC for this sample. That is to simplify it as much as it possible and remove all optional features.